libnmstate/nmstate.py

```diff
@@ -166,17 +167,47 @@ def _get_interface_info_from_plugins(plugins):
         for iface in plugin.get_interfaces():
             iface[IFACE_PRIORITY_METADATA] = plugin.priority
             iface_name = iface[Interface.NAME]
-            if iface_name in all_ifaces:
-                existing_iface = all_ifaces[iface_name]
+            iface_type = iface.get(Interface.TYPE, InterfaceType.UNKNOWN)
+            iface_index = f"{iface_type}.{iface_name}"
+            if iface_index in all_ifaces:
+                existing_iface = all_ifaces[iface_index]
                 existing_priority = existing_iface[IFACE_PRIORITY_METADATA]
                 current_priority = plugin.priority
                 if current_priority > existing_priority:
                     merge_dict(iface, existing_iface)
-                    all_ifaces[iface_name] = iface
+                    all_ifaces[iface_index] = iface
                 else:
                     merge_dict(existing_iface, iface)
             else:
-                all_ifaces[iface_name] = iface
+                all_ifaces[iface_index] = iface
+
+    # For any unknown interface, we merge it when there is only one interface
+    # with the same name and valid type
+    to_be_removed_index = []
```

Comment:
This section is a bit complected and merits its own function and a reasoning.

In addition:
- Why is this needed? What does it solve etc...
  (I think it is because the UNKNOWN ones are reported by the ovsdb plugin for system interfaces, so the idea is to overwrite an UNKNOWN type with a known type)
  Anyway, it requires documentation as this is a generic rule and it is not trivial.
- Can it be unit tested, covering all the flow paths?

Reply:
This is need for ovsdb plugin to include extra information for system interface like eth1. It's hard not waste of time for ovsdb plugin to detect the interface type of a ovs system interface.

The third party plugin might don't want to detect the interface type neither. But doing above complex guess in nmstate top level, we easy the plugin coder's life.

Reply:
I will add the unit test cases for this.